apis/groups/v1alpha1/group_types.go

```diff
@@ -199,29 +204,55 @@ type LDAPGroupLink struct {
 }
 
 // SharedWithGroups represents a GitLab Shared with groups.
+// At least one of the fields [GroupID, GroupIDRef, GroupIDSelector] must be set.
 type SharedWithGroups struct {
-	GroupID          int          `json:"groupId"`
-	GroupName        string       `json:"groupName"`
-	GroupFullPath    string       `json:"groupFullPath"`
-	GroupAccessLevel int          `json:"groupAccessLevel"`
-	ExpiresAt        *metav1.Time `json:"expiresAt"`
+	// The ID of the group to share with.
+	// +optional
+	// +crossplane:generate:reference:type=github.com/crossplane-contrib/provider-gitlab/apis/groups/v1alpha1.Group
+	GroupID *string `json:"groupId,omitempty"`
```

Comment:
Same as above, I feel that this should be an `*int` as well with a manual referencer implementation.